### PR TITLE
Fix `apex` version in Docker due to broken upstream

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Recently in NVIDIA Apex there has been a few changes including some new cpp extensions. These are causing our CI to fail. For now I've checked out a commit before the changes were introduced and install this version instead.

Fixes #<issue_number>

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]
nice Freeze :D

**Review by ananthsub — APPROVED** [review]

File: dockers/base-conda/Dockerfile [hunk 1]
```diff
@@ -116,6 +116,8 @@ RUN \
 RUN \
     # install NVIDIA AMP
     git clone https://github.com/NVIDIA/apex && \
+    # todo: hardcode Apex version as later commits break CI when cpp extensions are compiling
+    cd ./apex && git checkout 705cba91fc26413d484f5ce2983427d840199a97 && cd ../ && \
     pip install --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./apex && \
     rm -rf apex
```
Comment (carmocca) [inline dockers/base-conda/Dockerfile:122]:
```suggestion
    # todo: hardcode Apex version as later commits break CI when cpp extensions are compiling
    pip install --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" git+https://github.com/NVIDIA/apex@705cba9 
```

Same for the other change

**Comment by SeanNaren** [comment]
Thanks Carlos, much cleaner! Let's also wait till the metrics fix gets in to merge into here to see if we have passing tests as well

**Review by carmocca — APPROVED** [review]

**Comment by SeanNaren** [comment]
As @Borda mentioned seems fairscale has been forcing 1.8 for some builds: https://github.com/PyTorchLightning/pytorch-lightning/pull/7146/checks?check_run_id=2404986555#step:4:1797

This should be handled by #7017

**Pull request merged**